megatron/core/ssm/mamba_mixer.py

```diff
@@ -268,60 +268,68 @@ def __init__(
             )
 
         conv_dim = self.d_inner_local_tp + 2 * self.ngroups_local_tp * self.d_state  # x B C
-        with get_cuda_rng_tracker().fork():
-            # weight shape: [conv_dim, 1, d_conv]
-            # bias shape: [conv_dim]
-            self.conv1d = nn.Conv1d(
-                in_channels=conv_dim,
-                out_channels=conv_dim,
-                bias=conv_bias,
-                kernel_size=d_conv,
-                groups=conv_dim,
-                padding=d_conv - 1,
-                device=torch.cuda.current_device(),
-                dtype=config.params_dtype,
-            )
-            setattr(self.conv1d.weight, "tensor_model_parallel", True)
-            setattr(self.conv1d.bias, "tensor_model_parallel", True)
+        # weight shape: [conv_dim, 1, d_conv]
```

Comment:
This seems to be a pretty significant change to do this with a different rng tracker and not related to respsecting config.perform_initialization. Why this change?

Reply:
My understanding is that, there is no randomness in creating the conv layer if weights are not initialized, so the get rng tracker context manager should only apply to the param init (line 288)

Reply:
What if conv_bias=True?

This could also change randomness from previous versions since it'll use up numbers for the weight initialization even if it is overwritten below, so a run with seed=N from before this commit won't match a run with seed=N after this commit. Not sure if that's a big deal, but might unnecessarily change the golden values.

Reply:
I reverted this change now and opted to set the rng state in megatron bridge to workaround this issue.